### ISSUE TITLE
ERROR: Exception in ASGI application: IndexError: index 24 is out of bounds for axis 0 with size 24, and meanwhile the search hangs.

### ISSUE THREAD

**Issue by thuzhf** [aimhubio/aim#2624]
## 🐛 Bug

`aim up` server log:

<img width="797" alt="image" src="https://user-images.githubusercontent.com/8064242/228148700-34ac333c-ea71-41bd-a34c-56c47558fa1d.png">

Search hangs in aim web UI:
<img width="1004" alt="image" src="https://user-images.githubusercontent.com/8064242/228150223-c6e36706-085d-4ac7-905b-ffc0a0fb22ce.png">

### To reproduce

I am not sure how to reproduce exactly. Maybe if you run lots of experiments/runs and do search (no matter whether you search while the experiments are being run or after the experiments are being run) and it will occur.

### Expected behavior

The search does not hang. And the `aim up` server log does not have exceptions.

### Environment

- Aim Version (e.g., 3.0.1): 3.16.2 & 3.17.1
- Python version: 3.9.16
- pip version: 23.0.1
- OS (e.g., Linux): Linux
- Any other relevant information

**Comment by thuzhf** [comment]
@SGevorg What is blocked on my end?

**Comment by SGevorg** [comment]
ah I see, Aim hangs across all explorers on the UI?

**Comment by thuzhf** [comment]
I just tested all explorers, and found **only** metrics explorer hangs.

**Comment by SGevorg** [comment]
Thanks @thuzhf :-) 
Strange. Have you tracked things other than metrics?
Runs explorer is virtualized and auto-loads the rest of the runs if you scroll down (if it doesn't then that's a bug, but hope not haha).

**Comment by thuzhf** [comment]
@SGevorg 
1. Yes, I only tracked metrics.
2. Yes, runs explorer auto-loads the rest of the runs if I scroll down.

**Comment by thuzhf** [comment]
@gorarakelyan Do you mean steps like follows? I didn't specify these steps.
![image](https://user-images.githubusercontent.com/8064242/229365820-ec467abd-2763-4cd2-8406-d5d98ce3fad6.png)

**Comment by thuzhf** [comment]
> 

@SGevorg Today I encountered all explorers hung, and there's not any error logs from aim server.

**Comment by SGevorg** [comment]
this is definitely terrible news.
@mihran113 @gorarakelyan  @roubkar fyi

**Comment by roubkar** [comment]
@thuzhf does the browser page freeze or can you still use the UI (navigate to other pages from the sidebar, hit the cancel button, etc..)?
would like to understand whether you're facing a UI performance issue or there's no response from the server

**Comment by thuzhf** [comment]
@roubkar It does not freeze. I can still use the UI. It seems there's no response from the server.

**Comment by mihran113** [comment]
Hey @thuzhf! I've artificially recreated this case on my end. As the tracking call for `aim.Run` is not atomic, there can be a case when the step data is already written, and the process terminated before the value data was written, and this will result in an error mentioned above.  I've opened a pull request that will prevent such cases from happening in the future.

Regarding on how to fix your logs: 
I've pushed some changes to a branch named: `fix/temporary-fix-for-index-error`, you can install from the source of that branch using 
`pip install git+https://github.com/aimhubio/aim.git@fix/temporary-fix-for-index-error`
this will print the corrupted metric's run hash, metric name and context in the terminal of `aim up` command, then using that information you'll need to override the corrupted step via this simple script:

```
from aim import Run
run = Run(repo={path_to_your_repo}, run_hash={run_hash}, read_only=False)
run.track(0, name={metric_name}, context={context}, step=24)
```

Let me know if you encounter any issues during the process, will be glad to help.

**Comment by mihran113** [comment]
Closing the issue as the fix was shipped with `aim v3.17.4`.
